modules/ml/src/main/java/org/apache/ignite/ml/dataset/impl/cache/util/ComputeUtils.java

```diff
@@ -244,25 +287,24 @@ public static void removeData(Ignite ignite, UUID datasetId) {
             qry.setFilter(filter);
 
             C ctx;
-            UpstreamTransformerChain<K, V> chainCopy = Utils.copy(transformersChain);
-            chainCopy.modifySeed(s -> s + part);
+            UpstreamTransformer<K, V> transformer = transformerBuilder.build(env);
+            UpstreamTransformer<K, V> transformerCopy = Utils.copy(transformer);
```

Comment:
Could you please clarify why we need a copy here?

Reply:
See https://github.com/apache/ignite/pull/5484#discussion_r236280053